extension/src/time_weighted_average.rs

```diff
@@ -0,0 +1,422 @@
+
+use serde::{Serialize, Deserialize};
+use pg_sys::TimestampTz;
+use pg_sys::Datum;
+use std::slice;
+
+use flat_serialize::*;
+use pgx::*;
+use crate::{
+    aggregate_utils::{aggregate_mctx, in_aggregate_context},
+    flatten,
+    json_inout_funcs,
+    palloc::{Internal, in_memory_context}, pg_type
+};
+
+use time_weighted_average::{
+    tspoint::TSPoint, 
+    TimeWeightSummary,
+    TimeWeightError,
+    TimeWeightMethod,
+};
+
+// hack to allow us to qualify names with "timescale_analytics_experimental"
+// so that pgx generates the correct SQL
+mod timescale_analytics_experimental {
+    pub(crate) use super::*;
+    extension_sql!(r#"
+        CREATE SCHEMA IF NOT EXISTS timescale_analytics_experimental;
+    "#);
+}
+extension_sql!(r#"set search_path to 'timescale_analytics_experimental', 'public';"#);
```

Comment:
yea. I did that, I'm not horribly happy about it, because it's a real PITA and makes things much worse and more likely that we forget to do it correctly somewhere, but yeah, you could lose search_path modifications, this is an edge case definitely, because if you have search path modifications you made them when you connected and probably can make them again, but does seem nicer to not do that. I think this will eventually be addressed with an aggregate builder macro in pgx that will allow for schema qualification, which will be a much more elegant want to deal with this.

Reply:
I'd perfer to just keep the `set search_path; ... reset search_path` to a single `extension_sql!` block; I don't want to rely on pgx not doing something weird between them.

Reply:
(forgetting to add the schema seems to me like a non-issue right now, we test for that, and my experience adding deliberately-stable features is that the tests work annoyingly well)